File: 0036-valid-sudoku/0036-valid-sudoku.js
/**
 * @param {character[][]} board
 * @return {boolean}
 */
var isValidSudoku = function(board) {
    for(let i=0;i<9;i++){
        let row=new Set();
        let col=new Set();
        let box=new Set();
        for(let n=0;n<9;n++){
            let r=board[i][n];
            let c=board[n][i]
            let b=board[3*Math.floor(i/3)+Math.floor(n/3)][3*(i%3)+(n%3)]
            if(r!=='.'){
                if(row.has(r))return false;
                row.add(r)
            } 
            if(c!=='.'){
                if(col.has(c))return false;
                col.add(c)
            }
            if(b!=='.'){
                if(box.has(b)) return false;
                box.add(b)
            }
        }
    }
    return true;
};

File: 0036-valid-sudoku/0036-valid-sudoku.cs
public class Solution {
    public bool IsValidSudoku(char[][] board) {
        HashSet<char>[] row=new HashSet<char>[9];
        HashSet<char>[] col=new HashSet<char>[9];
        HashSet<char>[] box=new HashSet<char>[9];
        for(int i=0;i<9;i++){
            row[i] = new HashSet<char>();
            col[i] = new HashSet<char>();
            box[i] = new HashSet<char>();
        }
        for(int i=0;i<9;i++){
            for(int n=0;n<9;n++){
                char check=board[i][n];
                int b=(3*(i/3))+(n/3);
                if(check=='.') continue;
                if(!row[i].Add(check)) return false;
                if(!col[n].Add(check)) return false;
                if(!box[b].Add(check)) return false;

            }
        }
        return true;
    }  
}